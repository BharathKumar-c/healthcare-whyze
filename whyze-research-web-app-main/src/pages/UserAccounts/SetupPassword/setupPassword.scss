@import '../../../common_styles/common_variable';

.setup-password-component{
    display: flex;
    align-items: center;
    flex-direction: column;
    height: 100vh;
    background-image: url('../../../assets/EllipseTop.svg'), url('../../../assets/EllipseBottom.svg');
    background-position: left top, right bottom;
    background-repeat: no-repeat, no-repeat;
    background-size: 300px 300px;
    padding: 2.5rem;
    &-logo{
        display: flex;
        justify-content: space-around;
        align-items: center;
        padding: 1rem;
        font-size:1rem;
        font-weight: 700;
    }
    &-title{
        padding-top: 1rem;
        font-size: 26px;
        font-weight: 700;
    }
    &-description{
        font-size: 16px;
        padding-top: 0.5rem;
        padding-bottom: 1rem; 
    }
    &-input-box{
        display: flex;
        justify-content:space-around;
        flex-direction: column;
        padding: 0.5rem;
        width: 350px;
    }
    &-button-group{
        text-align: center;
        justify-content: center;
        display: flex;
        flex-direction: row;
        margin: 1.5rem;

        button {
            border-radius: 100px;
            font-size: 15px;
            font-weight: 500;
            text-transform: capitalize;
            padding: 0rem 5rem 0rem 5rem;
            height: 42px;
            width: 330px;
            ;
      
            &:nth-child(1) {
              background: $primary-color;
              color: white;
              border: none;
      
            }
        }
    }
    &-setup-acc{
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        font-weight: 400;
        font-size: 16px;
    }
}