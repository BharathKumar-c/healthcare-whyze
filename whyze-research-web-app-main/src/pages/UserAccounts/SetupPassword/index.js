import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { HidePass, ShowpassIcon, WhyzeLogo } from '../../../assets';
import { Button, Input } from '../../../components/basic';
import {
  errorConstants,
  labelAndTitleConstants,
  regexConstants,
} from '../../../constant';
import { authEndPoints } from '../../../constant/ApiEndPoints';
import ApiUtil from '../../../utils/ApiUtils';
import './setupPassword.scss';
import { RoutesUrls } from '../../../constant/RoutesUrls';

const SetupPassword = () => {
  const navigate = useNavigate();
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [passwordError, setPasswordError] = useState('');
  const [passwordIserror, setPasswordIserror] = useState(false);
  const [confirmpasswordiserror, setConfirmPassIserror] = useState(false);
  const [confirmPasswordError, setConfirmPasswordError] = useState('');
  const [passwordType, setPasswordType] = useState('password');
  const [confirmPasswordType, setConfirmPasswordType] = useState('password');

  const [isLoading, setIsLoading] = useState(false);
  const [queryToken, setQueryToken] = useState();

  useEffect(() => {
    const queryParameters = new URLSearchParams(window.location.search);
    const queryToken = queryParameters.get('token');
    setQueryToken(queryToken);
  }, []);
  const handleChange = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    if (name === 'password') {
      setPassword(value);
    } else if (name === 'confirmPassword') {
      setConfirmPassword(value);
    }
  };
  const confirmPasswordtoggle = () => {
    if (confirmPasswordType === 'password') {
      setConfirmPasswordType('text');
      return;
    }
    setConfirmPasswordType('password');
  };
  const togglePassword = () => {
    if (passwordType === 'password') {
      setPasswordType('text');
      return;
    }
    setPasswordType('password');
  };
  const handleProceed = async () => {
    if (password === '') {
      setPasswordError(errorConstants.passwordMustBeEntered);
      setPasswordIserror(true);
    } else if (password.length < 8) {
      setPasswordError(errorConstants.passwordLength);
      setPasswordIserror(true);
    } else if (!regexConstants.passwordRegex.test(password)) {
      setPasswordError(errorConstants.passwordMustContainReqCharacters);
      setPasswordIserror(true);
    } else {
      setPasswordError('');
      setPasswordIserror(false);
    }

    if (confirmPassword === '') {
      setConfirmPasswordError(errorConstants.passwordMustBeEntered);
      setConfirmPassIserror(true);
    } else {
      setConfirmPasswordError('');
      setConfirmPassIserror(false);
    }

    if (password !== confirmPassword) {
      setConfirmPasswordError(errorConstants.passwordsDoNotMatch);
      setConfirmPassIserror(true);
    } else {
      setIsLoading(true);
      setConfirmPasswordError('');
      setConfirmPassIserror(false);
      try {
        await ApiUtil.forgotPassword(authEndPoints.resetPassword, {
          password: password,
          token: queryToken,
        }).then(() => {
          setIsLoading(false);
          navigate(RoutesUrls.loginUrl, { replace: true });
        });
      } catch (error) {
        navigate('/', {
          state: {
            from: 'resetPassword',
            message: error.response.data.message,
          },
        });
        setIsLoading(false);
      }
    }
  };

  return (
    <div className="setup-password-component">
      <div className="setup-password-component-logo">
        <WhyzeLogo /> &nbsp;&nbsp;&nbsp;
        {labelAndTitleConstants.applicationName}
      </div>
      <div className="setup-password-component-title">
        {labelAndTitleConstants.createNewPassword}
      </div>
      <div className="setup-password-component-description">
        {labelAndTitleConstants.chooseStrongPassword}
      </div>
      <div className="setup-password-component-input-box">
        <Input
          lable={labelAndTitleConstants.password}
          type={passwordType}
          name={'password'}
          value={password}
          isError={passwordIserror}
          errorText={passwordError}
          suffix={
            passwordType === 'password' ? (
              <ShowpassIcon onClick={togglePassword} />
            ) : (
              <HidePass onClick={togglePassword} />
            )
          }
          handleInputChange={(e) => handleChange(e)}
        />
      </div>
      <div className="setup-password-component-input-box">
        <Input
          lable={labelAndTitleConstants.confirmPassword}
          type={confirmPasswordType}
          name={'confirmPassword'}
          value={confirmPassword}
          isError={confirmpasswordiserror}
          errorText={confirmPasswordError}
          suffix={
            confirmPasswordType === 'password' ? (
              <ShowpassIcon onClick={confirmPasswordtoggle} />
            ) : (
              <HidePass onClick={confirmPasswordtoggle} />
            )
          }
          handleInputChange={(e) => handleChange(e)}
        />
      </div>
      <div className="setup-password-component-button-group">
        <Button
          handleButtonClick={() => handleProceed()}
          label={labelAndTitleConstants.changePassword}
          size={'large'}
          isLoading={isLoading}
        />
      </div>
    </div>
  );
};

export default SetupPassword;
