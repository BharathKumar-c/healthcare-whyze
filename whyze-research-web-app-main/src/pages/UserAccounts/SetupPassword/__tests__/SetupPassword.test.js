import React from 'react';
import SetupPassword from '..';
import { fireEvent, render, waitFor } from '@testing-library/react';
import ApiUtil from '../../.././../utils/ApiUtils';
import { BrowserRouter } from 'react-router-dom';
const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

jest.mock('../../.././../utils/ApiUtils');

describe('SetupPassword component test', () => {
  test('SnapShot test', () => {
    const { container } = render(<SetupPassword />);
    expect(container).toMatchSnapshot();
  });
  test('password Input value changes', () => {
    const { container } = render(<SetupPassword />);
    const emailInput = container.querySelector('input[name="password"]');
    expect(emailInput.value).toMatch('');
    fireEvent.change(emailInput, { target: { value: 'Test@12345' } });
    expect(emailInput.value).toMatch('Test@12345');
  });

  test('compare two inputs with same value', () => {
    const { container } = render(<SetupPassword />);
    const paswwordInput = container.querySelector('input[name="password"]');
    const confirmPasswordInput = container.querySelector(
      'input[name="confirmPassword"]',
    );
    expect(paswwordInput.value).toMatch('');
    fireEvent.change(paswwordInput, { target: { value: 'Test@12345' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'Test@12345' } });
    expect(paswwordInput.value).toMatch(confirmPasswordInput.value);
  });

  test('User submit with empty password field', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SetupPassword />
      </BrowserRouter>,
    );
    const paswwordInput = container.querySelector('input[name="password"]');
    const confirmPasswordInput = container.querySelector(
      'input[name="confirmPassword"]',
    );
    const submitButton = getByRole('button');

    fireEvent.change(paswwordInput, { target: { value: '' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'Test@12345' } });
    fireEvent.click(submitButton);
    const passwordErrorField = getByText('Password must be entered');
    expect(passwordErrorField).toBeInTheDocument();
  });

  test('User submit with less then 8 charater password', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SetupPassword />
      </BrowserRouter>,
    );
    const paswwordInput = container.querySelector('input[name="password"]');

    const submitButton = getByRole('button');

    fireEvent.change(paswwordInput, { target: { value: '1234567' } });
    fireEvent.click(submitButton);
    const passwordErrorField = getByText(
      'Password must be at least 8 characters',
    );
    expect(passwordErrorField).toBeInTheDocument();
  });

  test('User submit without captial,small letter in  password', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SetupPassword />
      </BrowserRouter>,
    );
    const paswwordInput = container.querySelector('input[name="password"]');

    const submitButton = getByRole('button');

    fireEvent.change(paswwordInput, { target: { value: '123456789' } });
    fireEvent.click(submitButton);
    const passwordErrorField = getByText(
      'Password should contain uppercase, lowercase, number and symbol',
    );
    expect(passwordErrorField).toBeInTheDocument();
  });

  test('User submit with empty Confirmpassword field', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SetupPassword />
      </BrowserRouter>,
    );
    const passwordInput = container.querySelector('input[name="password"]');
    const confirmPasswordInput = container.querySelector(
      'input[name="confirmPassword"]',
    );
    const submitButton = getByRole('button');
    fireEvent.change(passwordInput, { target: { value: 'Test@12345' } });
    fireEvent.change(confirmPasswordInput, { target: { value: '' } });
    fireEvent.click(submitButton);
    const confirmpasswordErrorField = getByText(/Passwords do not match/i);

    expect(confirmpasswordErrorField).toBeInTheDocument();
  });
  test('test change password submit', async () => {
    const { container, getByRole } = render(
      <BrowserRouter>
        <SetupPassword />
      </BrowserRouter>,
    );
    const passwordInput = container.querySelector('input[name="password"]');
    const confirmPasswordInput = container.querySelector(
      'input[name="confirmPassword"]',
    );

    fireEvent.change(passwordInput, { target: { value: 'Test@12345' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'Test@12345' } });

    ApiUtil.forgotPassword.mockResolvedValueOnce({
      message:
        'You will receive a password recovery link at your email address in a few minutes If your email address exists in our system.',
    });
    const submitButton = getByRole('button');
    fireEvent.click(submitButton);
    await waitFor(() => {
      expect(ApiUtil.forgotPassword).toHaveBeenCalledWith('resetPassword', {
        password: 'Test@12345',
        token: null,
      });
      expect(ApiUtil.forgotPassword).toHaveBeenCalledTimes(1);
      expect(mockedUseNavigate).toHaveBeenCalledWith('/', { replace: true });
    });
  });
});
