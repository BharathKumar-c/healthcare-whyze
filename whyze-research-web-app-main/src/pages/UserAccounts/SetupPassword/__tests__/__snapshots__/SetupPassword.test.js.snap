// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPassword component test SnapShot test 1`] = `
<div>
  <div
    class="setup-password-component"
  >
    <div
      class="setup-password-component-logo"
    >
      <svg>
        WhyzeLogo.svg
      </svg>
      WHYZE
    </div>
    <div
      class="setup-password-component-title"
    >
      Create a new password
    </div>
    <div
      class="setup-password-component-description"
    >
      Choose a strong password
    </div>
    <div
      class="setup-password-component-input-box"
    >
      <div>
        <span
          class="ant-typography input-lable"
        >
          Password
        </span>
        <span
          class="ant-input-affix-wrapper"
          style="border-radius: 0px;"
        >
          <input
            class="ant-input"
            name="password"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <svg>
              ShowpassIcon.svg
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="setup-password-component-input-box"
    >
      <div>
        <span
          class="ant-typography input-lable"
        >
          Confirm Password
        </span>
        <span
          class="ant-input-affix-wrapper"
          style="border-radius: 0px;"
        >
          <input
            class="ant-input"
            name="confirmPassword"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <svg>
              ShowpassIcon.svg
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="setup-password-component-button-group"
    >
      <button
        class="ant-btn ant-btn-default ant-btn-lg"
        type="button"
      >
        <span>
          Change Password
        </span>
      </button>
    </div>
  </div>
</div>
`;
